src/features/users/user-controller.ts

```diff
@@ -436,17 +436,19 @@ class UserController {
         };
     }
 
-    async verifyUser(verifyToken: string): Promise<boolean> {
+    async verifyUser(verifyToken: string, confirmEmail: string): Promise<boolean> {
         const user = await this.getUserByVerifyToken(verifyToken);
         if (_.isNil(user?.verifyToken)) {
             throw new IllegalArgumentException('Invalid verification token');
-        } else if(user.verifyToken !== verifyToken) {
+        } else if (user.verifyToken !== verifyToken) {
             throw new IllegalArgumentException('Invalid verification token');
-        } else if(moment().isAfter(user.verifyTokenExpiresAt)) {
+        } else if (moment().isAfter(user.verifyTokenExpiresAt)) {
             throw new IllegalArgumentException('Verification token has expired');
-        } else if(user.verified) {
+        } else if (user.verified) {
             logger.warn('Verification token should be set to null on verify, thus this should not be possible');
             throw new IllegalArgumentException('Already verified');
+        } else if (confirmEmail !== user.email) {
```

Comment:
I personally prefer keeping things like this obscure for security reasons but I'm ok with it

I would have just added this to getUserByVerifyToken so it would return null if the combo was not correct